src/basic/process-util.c

```diff
@@ -128,14 +129,21 @@ int get_process_cmdline(pid_t pid, size_t max_length, bool comm_fallback, char *
                                 return -ENOMEM;
                         }
 
-                        r[len++] = isprint(c) ? c : ' ';
-                }
+                        if (isprint(c)) {
+                                if (space) {
+                                        r[len++] = ' ';
+                                        space = false;
```

Comment:
if there are two characters added to the string in a single loop, then the GREEDY_REALLOC() call above needs to be allocated for "len+3" to have enough space for the trailing NUL...